cmd/doctor.go

```diff
@@ -0,0 +1,191 @@
+package cmd
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"path/filepath"
+	"strings"
+
+	"github.com/exercism/cli/api"
+	"github.com/exercism/cli/config"
+	"github.com/exercism/cli/workspace"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+)
+
+// doctorCmd reports workspace cleanup tasks.
+var doctorCmd = &cobra.Command{
+	Use:     "doctor",
+	Aliases: []string{"doc"},
+	Short:   "Doctor reports workspace cleanup tasks.",
+	Long: `Doctor reports workspace cleanup tasks.
+
+	Use --fixup to run reported tasks.
+`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		cfg := config.NewConfig()
+
+		usrCfg := viper.New()
+		usrCfg.AddConfigPath(cfg.Dir)
+		usrCfg.SetConfigName("user")
+		usrCfg.SetConfigType("json")
+		// Ignore error. If the file doesn't exist, that is fine.
+		_ = usrCfg.ReadInConfig()
+		cfg.UserViperConfig = usrCfg
+
+		v := viper.New()
+		v.AddConfigPath(cfg.Dir)
+		v.SetConfigName("cli")
+		v.SetConfigType("json")
+		// Ignore error. If the file doesn't exist, that is fine.
+		_ = v.ReadInConfig()
+
+		return runDoctor(cfg, cmd.Flags())
+	},
+}
+
+func runDoctor(cfg config.Config, flags *pflag.FlagSet) error {
+	usrCfg := cfg.UserViperConfig
+	if usrCfg.GetString("workspace") == "" {
+		return fmt.Errorf(msgRerunConfigure, BinaryName)
+	}
+
+	fixup, err := flags.GetBool("fixup")
+	if err != nil {
+		return err
+	}
```

Comment:
In terms of design, what I think would help make it easy to do both modes (reporting and fixing), would be to have a type for each type of thing we're fixing. We can initialize the value, and have three different steps:

1. Investigate.
2. Report.
3. Fix.

The investigation would store whatever it needs to make it easy to both report and fix.